### PR TITLE
Issue 19060

### PR DESCRIPTION
**Pull request by erickgonzalez**
This PR incorporates the amazon Rekognition actionlet to core. 

It can be configured using the Apps.

When an error is reached executing the actionlet it should be logged an move on, no error should be thrown.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rekognition/api/RekognitionAPI.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import com.amazonaws.services.rekognition.model.DetectLabelsRequest;
+import com.amazonaws.services.rekognition.model.DetectLabelsResult;
+import com.amazonaws.services.rekognition.model.Image;
+import com.amazonaws.services.rekognition.model.Label;
+import java.util.stream.Collectors;
+
+public class RekognitionAPI {
+
+
+  private final AWSCredentials awsCredentials;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rekognition/api/RekognitionAPI.java:25]: Codacy found an issue: [Perhaps 'awsCredentials' could be replaced by a local variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6017382)

File: dotCMS/src/main/java/com/dotcms/rekognition/api/RekognitionAPI.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                    .standard()
+                    .withRegion(Regions.US_WEST_2)
+                    .withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
+                    .build();
+
+  }
+
+  public List<String> generateTags(final File file, final int maxLabels, final float minConfidence) throws IOException {
+
+    try (final RandomAccessFile aFile = new RandomAccessFile(file.getAbsolutePath(), "r")) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rekognition/api/RekognitionAPI.java:42]: Codacy found an issue: [Unnecessary modifier 'final' on resource specification 'aFile': resource specifications are implicitly final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6017382)

File: dotCMS/src/main/java/com/dotcms/rekognition/actionlet/RekognitionActionlet.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            awsTags.add(TAGGED_BY_AWS);
+            for (final String tag : awsTags) {
+                tagAPI.addContentletTagInode(tag, contentlet.getInode(), contentlet.getHost(), tagFieldOpt.get().variable());
+            }
+
+            HibernateUtil.addCommitListener(contentlet.getInode(),
+                    ()->this.refresh(contentlet));
+
+            CacheLocator.getContentletCache().remove(contentlet);
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rekognition/actionlet/RekognitionActionlet.java:136]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6017382)

File: dotCMS/src/main/java/com/dotcms/rekognition/api/RekognitionAPI.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+      final FileChannel inChannel = aFile.getChannel();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rekognition/api/RekognitionAPI.java:44]: Codacy found an issue: [Ensure that resources like this FileChannel object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6017382)

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]
Pretty cool

**Pull request merged**